runtime/java/src/common/RTTIManager.java

```diff
@@ -0,0 +1,84 @@
+package common;
+
+import java.util.*;
+
+/**
+ * Collects maps between fully-qualified silver names (some:package:NT) and singletons for terminals,
+ * nonterminals, and productions. The singletons represent the terminal/nonterminal/production itself,
+ * not instances thereof. So functions on the singletons grant access to static fields/methods on the
+ * terminal/nonterminal/production. These are meant to be usable for all the same functions we
+ * previously did via java reflection. So a Terminalton<T> is akin to a Class<T extends Terminal>, a
+ * Nonterminalton<T> to a Class<T extends Node>, and Prodleton<T> to a Class<T extends N extends Node>
+ * where .getNonterminalton() -> Nonterminalton<N super T>. These are registered into these lookup
+ * tables at silver init time.
+ * 
+ * This is used for the following:
+ *  - silver reflection
+ *  - decorators (read: implementing autocopy)
+ * 
+ * @author louisg
+ */
+public final class RTTIManager {
```

Comment:
I'm still not thrilled about the name `RTTIManager` ...but I also don't have a better one.